src/componenten/counter/counter.js

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+
+function counter({fruitName, hook, setHook}) {
+    return (
+        <article>
+            <h2> {fruitName}</h2>
+            <div>
+                <button type={"button"}
+                        onClick={() => setHook(hook - 1)}
+                        disabled={hook === 0}>
+                    -
+                </button>
+                <span>{hook}</span>
+                <button
+                    type="button"
+                    onClick={() => setHook(hook + 1)}>
```

Comment:
Nu zal telkens wanneer `hook` veranderd een onnodige rerender van deze functie plaatsvinden. Wat je zou kunnen doen is de huidige waarde gebruiken die je vanuit de setHook functie al krijgt:
```suggestion
                    onClick={() => setHook(currentValue => currentValue + 1)}>
```